feat: dont beat around the fees bush

gomesalexandre opened the pull request:
## Description

This PR simplifies the fees by only telling users about the bps as a percentage if applicable, on rate click.
While at it:

- Removes the protocol line items to match mocks
- Makes limit not expandable at input (there's effectively nothing remaining)
- Uses the same rate/gas component for trade confirm as in input for consistency, and ensures no duplicate "Rate" line item
- Rename "Transaction Fee" to "Network Fee" and add a tooltip explainer
- Fixes the dp fixing of the rate to be in the buy asset precision (this is a bug currently present in all envs where the rate would erroneously show more digits than said token has for the rate)

```
- [x] remove fees line items
- [x] discuss with @twblack88 where popover/trigger
- [x] Match rate/gas row with input and remove protocol line item 
- [x] Check all other consumoors (limit, LP, RFOX) have happy UI
```

## Issue (if applicable)

<!-----------------------------------------------------------------------------
If applicable, please link to the github issue and put `closes #XXXX` in your comment to auto-close the issue that your PR fixes.
------------------------------------------------------------------------------>

closes https://github.com/shapeshift/web/issues/8800
closes https://github.com/shapeshift/web/issues/9008

## Risk

> High Risk PRs Require 2 approvals

<!-----------------------------------------------------------------------------
Outline the scope of your changes and the risk associated with them. You must use your discretion as an engineer to determine the potential impact of your changes.

WARNING: If your PR introduces a new on-chain transaction or modifies an existing one, please add the "High-Risk" label to this PR and ask for 2 reviewers to approve it before merging.

E.g. an upgrade to `hdwallet` or core state management would be considered higher risk, and might require a full regression test. UI or isolated view changes, or something behind a feature flag may have near zero risk. Small bug fixes might require testing isolated to the specific fix.
------------------------------------------------------------------------------>

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

## Testing

<!-----------------------------------------------------------------------------
We treat every PR to be merged with the same scrutiny as if we were merging directly to production.

Your PR will not be merged if you do not complete the sections below for our engineering and operations teams to test.
------------------------------------------------------------------------------>

- Spot input and confirm summary looks good
- Limit input and confirm summary looks good

### Engineering

<!-----------------------------------------------------------------------------
Include sufficient information here for an engineer to test your PR. This may include how to test locally, in a built environment, changes to infrastructure etc.
------------------------------------------------------------------------------>

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

<!-----------------------------------------------------------------------------
If your changes have a user-facing impact, describe how a non-technical QA team can functionally test your changes in a preview environment.

If they are not user-facing please describe how to test for any regressions that may occur.
------------------------------------------------------------------------------>

## Screenshots (if applicable)

https://jam.dev/c/471afcb3-cfdb-4fb9-af0e-8156f5382a1c

gomesalexandre commented:
Converting to draft as there seems to be some misalignment here re:
- whether we want to use the rate/gas row consistently (seems like consistency is key, as we already use it in input. Proposed by @twblack88 and implemented in this PR, but not part of original designs)
- or, instead of using the rate/gas row in confirm screen, just have items there, unexpanded. Meaning we'd bring a rate line item (which isn't here in this PR, as the rate/gas row is a drop-in replacement for it)

Happy with whichever solution we go with, though if we go with the second, feels odd to have a rate/gas row in trade input and none in the trade confirm screens, something something consistency? Whichever way we go with, should look the same in both screens IMO.

twblack88 commented:
- Add tooltips on the gas icon and the rate so it's clear what that means if someone is moving fast
- make the information expanded by default

did i miss anything from grooming @0xApotheosis @gomesalexandre ?

0xApotheosis commented:
Only expanded by default on the confirm screen though, right @twblack88? Not the preview screen.

twblack88 commented:
Yep expanded on confirm, that's where we aligned IIRC.

NeOMakinG reviewed (approved):
## Styling hot takes
<img width="486" alt="image" src="https://github.com/user-attachments/assets/4648dd89-7622-4a48-8fd3-5f1c1407888e" />

The card footer is missing some padding, the submit button is touching the borders

https://jam.dev/c/bdbd8845-5925-4178-bc7d-a18cf76be3c7

The rate blue styling with underline makes me think that there is something cliquable (as well as the double arrows), while nothing happens on click

Those thoughts are also present inside the asset page swapper (except for the blue underlined weird styling)

https://jam.dev/c/e1aff2a0-11fa-4563-9235-7fa240b6d988

I'm pre stamping because from a functional point of view, it looks fine, but please massage those part a little bit 🙏

gomesalexandre commented:
> The card footer is missing some padding, the submit button is touching the borders

Nice catch! Fixed in 81139c6ffa:

<img width="548" alt="image" src="https://github.com/user-attachments/assets/8836e8ef-eb8b-4f22-a1ee-4b1378ef0076" />
<img width="563" alt="image" src="https://github.com/user-attachments/assets/98022fd5-b1f0-46cc-9d49-5af9f878bd58" />
<img width="576" alt="image" src="https://github.com/user-attachments/assets/afdaa594-a3b0-4d29-80a3-4c8e45a570c2" />
<img width="552" alt="image" src="https://github.com/user-attachments/assets/ec18c7ce-8277-4ab7-93dd-48bce0d616e4" />
<img width="565" alt="image" src="https://github.com/user-attachments/assets/ca6b6147-5f81-4424-ba81-5a75dac71c19" />

> The rate blue styling with underline makes me think that there is something cliquable (as well as the double arrows), while nothing happens on click

Fixed in 81139c6ffa as well 
<img width="558" alt="image" src="https://github.com/user-attachments/assets/c83a6b44-30f1-4752-b27f-482251954a1d" />

gomesalexandre commented on src/components/MultiHopTrade/components/RateGasRow.tsx:
```diff
@@ -45,9 +66,39 @@ export const RateGasRow: FC<RateGasRowProps> = memo(
+    const feeMessage = useMemo(() => {
+      if (!affiliateBps || bnOrZero(affiliateBps).isZero())
+        return translate('trade.freeTrade', {
+          foxBalance: bnOrZero(foxBalanceCryptoPrecision).toFixed(2),
```
Copied over from https://github.com/shapeshift/web/pull/8953/files#diff-ba71cac3885a1f4a986d3c3bffc2a5a4cefc5649bfacc8192ec80414ee8b89d4L95 but yeah I'd say we better not display any dp altogether given FOX price.

gomesalexandre replied:
42930c2f3afb2a682a99eb3cb8c5580575b8c005

<img width="322" alt="image" src="https://github.com/user-attachments/assets/44a7de54-9ad7-4630-8e66-2fdbbef362b3" />
<img width="360" alt="image" src="https://github.com/user-attachments/assets/6a4b3e28-9a86-40a7-8dba-96dc19ebf0de" />

gomesalexandre commented on src/components/MultiHopTrade/components/LimitOrderV2/LimitOrderDetail.tsx:
```diff
@@ -26,47 +26,32 @@ export const LimitOrderDetail = () => {
-        </Row.Label>
-        <Row.Value textAlign='right'>
+    <RateGasRow
+      affiliateBps={undefined}
```
Ohh nice catch, fixed in ed817f9239:

<img width="418" alt="image" src="https://github.com/user-attachments/assets/7e78b703-ee6c-43fe-a0d2-e871c0f746cc" />
<img width="519" alt="image" src="https://github.com/user-attachments/assets/805349af-dc92-4ffe-a548-ee89d3b12d2a" />
